packages/webview_flutter/ios/Classes/FlutterWebView.m

```diff
@@ -347,4 +352,29 @@ - (void)registerJavaScriptChannels:(NSSet*)channelNames
   }
 }
 
+- (void)onGetUserAgent:(FlutterMethodCall*)call result:(FlutterResult)result {
+  [_webView
+      evaluateJavaScript:@"navigator.userAgent"
```

Comment:
I just realized we can't rely on JavaScript here as it may be disabled.
Sorry for missing this in previous iterations.

I guess if WkWebView doesn't support getting the user agent we should probably let our users inject js to get it as they are the ones in control of the js policy.

My suggestion for this PR would be to not include the `getUserAgent` method, and inject js to get the user agent in the e2e tests.

We should file an issue for `getUserAgent` if that turns out to be important to many users we should consider adding it as an Android only API (maybe as part of the Android controller).

Reply:
I see, it makes sense that we shouldn't rely on injected javascript to get the user agent in the WkWebView. As the main goal of this pr was to add the ability to set the user agent, I don't mind removing the `getUserAgent` method. I removed all methods involved in getting the user agent and updated the tests accordingly.